BooruSharp.Others/Pixiv.cs

```diff
@@ -27,76 +27,40 @@ public Pixiv()
         {
-            if (username == null)
-                throw new ArgumentNullException(nameof(username));
-
-            if (password == null)
-                throw new ArgumentNullException(nameof(password));
-
-            var request = new HttpRequestMessage(HttpMethod.Post, "https://oauth.secure.pixiv.net/auth/token");
-
-            string time = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ss+00:00");
-            request.Headers.Add("X-Client-Time", time);
-
-            using (var md5 = MD5.Create())
+            public SessionInfo(string accessToken, string refreshToken, DateTime expirationDate)
             {
-                var hashBytes = md5.ComputeHash(Encoding.UTF8.GetBytes(time + _hashSecret));
-                var hashString = BitConverter.ToString(hashBytes).Replace("-", null).ToLowerInvariant();
-
-                request.Headers.Add("X-Client-Hash", hashString);
+                AccessToken = accessToken ?? throw new ArgumentNullException(nameof(accessToken));
+                RefreshToken = refreshToken ?? throw new ArgumentNullException(nameof(refreshToken));
+                ExpirationDate = expirationDate;
             }
 
-            request.Content = new FormUrlEncodedContent(
-                new Dictionary<string, string>
-                {
-                    { "get_secure_url", "1" },
-                    { "client_id", _clientID },
-                    { "client_secret", _clientSecret},
-                    { "grant_type", "password" },
-                    { "username", username },
-                    { "password", password }
-                });
+            public string AccessToken { get; }
 
-            var response = await HttpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
-
-            if (response.StatusCode == HttpStatusCode.BadRequest)
-                throw new AuthentificationInvalid();
-
-            response.EnsureSuccessStatusCode();
+            public string RefreshToken { get; }
 
-            var jsonToken = JsonConvert.DeserializeObject<JToken>(await response.Content.ReadAsStringAsync());
-            var responseToken = jsonToken["response"];
-
-            AccessToken = responseToken["access_token"].Value<string>();
-            RefreshToken = responseToken["refresh_token"].Value<string>();
-            _refreshTime = DateTime.Now.AddSeconds(responseToken["expires_in"].Value<int>());
+            public DateTime ExpirationDate { get; }
         }
 
+        /// <inheritdoc/>
+        public override bool IsSafe => false;
+
         /// <summary>
         /// Sends a login API request using specified refresh token.
         /// </summary>
         /// <returns>The task object representing the asynchronous operation.</returns>
         /// <exception cref="AuthentificationInvalid"/>
         /// <exception cref="HttpRequestException"/>
-        public async Task LoginAsync(string refreshToken)
+        public Task LoginAsync(string refreshToken)
```

Comment:
Not sure if it's worth keeping this method at all. Authorization through refresh token is now implemented internally and I don't see anyone using their refresh token to log in at all.

Reply:
Authentification by refresh token allow someone to just reuse his token, or take it from somewhere else and login in BooruSharp using it